File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        seen = {}
        for i in range(n):
            if nums[i] in seen:
                if i - seen[nums[i]] <= k:
                    return True
            seen[nums[i]] = i
        
        return False 

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> seen;
        int left = 0;
        int maxcnt = 0;
        for(int i =0; i< s.length(); i++){
            while(seen.find(s[i]) != seen.end()){
                seen.erase(s[left]);
                left++;
            }

            seen.insert(s[i]);

            int cnt = i- left + 1;

            maxcnt = max(maxcnt, cnt);
        }

        return maxcnt;
    }
};
